syntax = "proto3";

service Order{
    rpc get_order_by_id (OrderRequest) returns (OrderResponse);
}

message OrderRequest {
    string order_id = 1;
}

message OrderResponse {
    string id = 1;
    string customer_id = 2;
    repeated OrderLine order_lines = 3;
}

message OrderLine {
    string id = 1;
    string order_id = 2;
    string product_id = 3;
    double product_price = 4;
}
